Hadrons/Modules/MDistil/DistilMesonField.hpp

```diff
@@ -219,8 +236,49 @@ void TDistilMesonField<FImpl>::execute(void)
     std::map<std::string, DistilVector & > distVectors = {{"left",dvl}  ,{"right",dvr}};
     DistillationNoise &noisel = envGet( DistillationNoise , par().leftNoise);
     DistillationNoise &noiser = envGet( DistillationNoise , par().rightNoise);
-    std::map<std::string, DistillationNoise & >   noises = {{"left",noisel},{"right",noiser}};
-    
+
+    //auxiliar lambda functions
+    auto filenameDmfFn = [this, outpath](const unsigned int m, const unsigned int o, const int nl, const int nr)
+    {
+        std::stringstream ss;
+        ss << gamma_[o] << "_p";
+        for (unsigned int mu = 0; mu < momenta_[m].size(); ++mu)
+            ss << momenta_[m][mu] << ((mu == momenta_[m].size() - 1) ? "" : "_");
+
+        std::string filename = outpath + ss.str(); 
+        if(!isExact_)
```

Comment:
ok, I don't fully understand this isExact. So only if both fields are stochastic you write that into their filenames?

Reply:
This means that I just write the noise pair to the name of the file if the evaluation is stochastic (in exact distil the noise pair doesn't have any significance and is always 0_0. Again, assuming no mix between exact and stochastic